plugins/powerd/fu-plugin-powerd.c

```diff
@@ -98,6 +98,10 @@ fu_plugin_update_prepare (FuPlugin *plugin,
 		       &current_state,
 		       &current_level);
 
+	/* permitting updates on devices that run only on external power */
+	if (current_state == FU_BATTERY_STATE_UNKNOWN && current_level == FU_BATTERY_VALUE_INVALID)
```

Comment:
I believe current_level == -1 in powerd while FU_BATTERY_VALUE_INVALID is 101

Reply:
Instead of doing this:

Add to this change the additional check on https://github.com/fwupd/fwupd/commit/ab60cc88f4398f1f4c6a5a6feb845ab6ae8da7e1#diff-bab80cfcd60395be86ab4adf78550140d0eeec0a59237403bf0aca667afb17b3

And substitute this for: 
if (current_level == -1)
  current_level = FU_BATTERY_VALUE_INVALID;

This eventually would be moved out to the "timer method" similar to https://github.com/fwupd/fwupd/blob/master/plugins/upower/fu-plugin-upower.c#L60

Reply:
Actually this two additional checks below to line 112 :

```
fu_context_get_battery_level (ctx) != FU_BATTERY_VALUE_INVALID &&
fu_context_get_battery_threshold (ctx) != FU_BATTERY_VALUE_INVALID
```